scripts/device-common-install/libexec/device_hw_validation.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+is_expected_hdd() {
+  local expected_value
+  local tolerance
+
+  expected_value="${1?ERROR variables is unset}"
+  tolerance="${2?ERROR variable is unset}"
+
+  hdd_lower_bound=$(expr "$1" - "$2")
```

Comment:
Shouldn't this use bc for the same reason  as below?

Also, `"${expected_value} - ${tolerance}"`